### PR TITLE
Abjurant champion

### PR DESCRIPTION
**Pull request by doug1234**
Draft of the abjurant champion prestige class.

The one issue I have is that the arcane boost action brings up the picker.  The only thing that can be selected is the character that is preforming the action (which is what I want).  Still it would be better if the picker was not displayed at all.

### CODE REVIEW

File: tpdatasrc/tpgamefiles/rules/d20_actions/action02900_abjurant_champion_arcane_boost.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import tpactions
+
+def GetActionName():
+	return "Arcane Boost"
+
+def GetActionDefinitionFlags():
+	return D20ADF_None
+	
+def GetTargetingClassification():
+	return D20TC_CastSpell
```
Comment (DudeMcDude) [inline tpdatasrc/tpgamefiles/rules/d20_actions/action02900_abjurant_champion_arcane_boost.py:11]: So why do you actually need this to be d20tc_castspell?

Comment (doug1234) [reply tpdatasrc/tpgamefiles/rules/d20_actions/action02900_abjurant_champion_arcane_boost.py:11]: I need the spell packet for getting the spell level and debiting the spell.  If I do D20TC_Target0 (which would not show targeting), I don't get a spell packet.

Comment (DudeMcDude) [reply tpdatasrc/tpgamefiles/rules/d20_actions/action02900_abjurant_champion_arcane_boost.py:11]: Well that data is set according to what's already stored in evt_obj.d20a.spell_data, so you could just retrace it by creating a tpdp.SpellPacket(), setting its fields just like D20::TargetCheck() does and then calling debit from it. If there isn't already you should add a helper method for creating a SpellPacket from D20SpellData.

Comment (doug1234) [reply tpdatasrc/tpgamefiles/rules/d20_actions/action02900_abjurant_champion_arcane_boost.py:11]: Thanks!  That is an interesting idea.  Looks like a spell packet can currently only be created from a spell id but I'll see about making a way to construct one from D20SpellData.

Comment (doug1234) [reply tpdatasrc/tpgamefiles/rules/d20_actions/action02900_abjurant_champion_arcane_boost.py]: That seemed to work great.  I pushed up the change.  I'll test just a bit more and remove the draft tag on Friday.

**Comment by DudeMcDude** [comment]
Awesome, so is it ready for merge?

**Comment by doug1234** [comment]
It is done but I wanted to test it a bit more tomorrow night.  Then I will remove the draft tag.

**Comment by doug1234** [comment]
Should be ready now.

**Pull request merged**